utils/configuration.go

```diff
@@ -76,7 +79,52 @@ func ParseServerTLS(configuration *viper.Viper, tlsRequired bool) (*tls.Config,
 		}
 	}
 
-	return tlsconfig.Server(tlsOpts)
+	// Fetch TLS config as usual
+	conf, err := tlsconfig.Server(tlsOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	// Reload certificate hourly
+	var reloadCert *tls.Certificate
+	var m sync.RWMutex
+	var resetCert func()
+	resetCert = func() {
+		m.Lock()
+		defer m.Unlock()
+		reloadCert, err = loadCertificate(tlsOpts.CertFile, tlsOpts.KeyFile)
+		if err != nil {
+			logrus.Warnf("failed to (re)load certificate %s", err)
+		}
+		// Re-run this function hourly
+		time.AfterFunc(time.Hour, resetCert)
```

Comment:
The cadence seems like it should be configurable.

Reply:
Default 0 == no rotation works for me. It would be unreasonable to interpret it as "always be rotating" so there shouldn't be any future conflict :)